tests/rptest/tests/consumer_group_recovery_tool_test.py

```diff
@@ -44,7 +44,19 @@ def __init__(self, test_ctx, *args, **kwargs):
     def describe_all_groups(self):
         rpk = RpkTool(self.redpanda)
         all_groups = {}
-        for g in rpk.group_list():
+
+        def do_list_groups():
+            res = rpk.group_list()
+            assert res is not None
+            return False if len(res) == 0 else (True, res)
```

Comment:
it'd be a bit of a stronger guarantee if you took as a parameter the expected number of groups rather than waiting for a non-zero number of groups. the reasoning being that the race we are trying to deal with seems like it could result in a partial result, not necessarily all or nothing.

Reply:
done.